### PR TITLE
Implement form input components

### PR DESCRIPTION
**Pull request by sarahhjchung**
## Changes
- Implemented the `TextFormInput` and `DropdownMenuInput` components

## Notes
- I slightly modified the parameters for these components slightly to adjust for the Material3 components I used 
- Below I attached a recording of what the TextFormInput and DropdownMenuInput look like.

[form-input-recording-1.webm](https://user-images.githubusercontent.com/60327675/233194252-d66ca8ac-a831-4f00-9222-4f560be99f4a.webm)

### CODE REVIEW

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/components/TextFormInput.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    required: Boolean = false,
+    errorMessage: String = "",
+) {
+    val label = if (required) "$labelText *" else labelText
+    var text by remember { mutableStateOf(value) }
+    var isError by remember { mutableStateOf(false) }
+
+    fun validate(text: String) {
+        isError = (required && text == "")
+    }
```
Comment (sarahhjchung) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/components/TextFormInput.kt]: I'm not sure if these should go in the component or if this logic will be in the screen. Please advise what the best way should be

Comment (sarahhjchung) [reply app/src/main/kotlin/com/twain/interprep/presentation/ui/components/TextFormInput.kt]: @sarahhjchung move validate function to utils class

**Comment by Audienix** [comment]
@sarahhjchung Is component text style matching with figma ? Can you add the screenshot in PR description? Also how we are using Dropdown component? Haven't seen it's usage.

**Review by YANLING-H — COMMENTED** [review]
It is recommended to add a new line at the end of each file.

**Comment by sarahhjchung** [comment]
@Audienix @YANLING-H The reason why the text style behaviour was different from Figma is because I overrode the type for `bodyLarge` in `Type.kt` for the FullScreenEmptyState component because the texts in that component had different font sizes and weights than the given Material3 typography styles. So overridden `bodyLarge` was affecting the TextFormInput component because the default text style for OutlinedTextFields are `bodyLarge`. So I updated `Type.kt` and the FullScreenEmptyState component and now the text style matches Figma

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/components/TextFormInput.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            } else if (text.isNotEmpty()) {
+                IconButton(
+                    onClick = { /* TODO: implement onClick for clearing text */ }
+                ) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.cancel_icon),
+                        contentDescription = "Cancel"
+                    )
+                }
+            }
```
Comment (sarahhjchung) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/components/TextFormInput.kt:68]: Added the clear button here

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt [hunk 3]
```diff
@@ -95,7 +95,33 @@ fun DashboardScreen(
                         fontWeight = FontWeight.Normal
                     )
                 }
-                items(25) { PastInterviewCard() }
+                items(25) {
+                    Column(
```
Comment (Arighna88) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/modules/dashboard/DashboardScreen.kt]: This was supposed to be only for testing purpose not for committing here. PasInterViewCard() should be shown here.

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/components/TextFormInput.kt [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TextFormInput(
+    modifier: Modifier,
+    value: String = "",
+    labelText: String,
+    bottomText: String = "",
+    required: Boolean = false,
+    errorMessage: String = "",
```
Comment (Arighna88) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/components/TextFormInput.kt]: Change it to errorText for uniformity in naming ?

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/components/DropdownMenuInput.kt [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+@Composable
+fun DropdownMenuInput(
+    modifier: Modifier,
+    options: List<String>,
+    labelText: String,
+) {
+    var expanded by remember { mutableStateOf(false) }
+    var selectedOptionText by remember { mutableStateOf("") }
+
+    ExposedDropdownMenuBox(
```
Comment (Arighna88) [inline app/src/main/kotlin/com/twain/interprep/presentation/ui/components/DropdownMenuInput.kt:19]: In the textinput fields where dropdown is shown , user should also be allowed to type. Are we allowing now?

**Review by Arighna88 — APPROVED** [review]
Approved to unblock. But left few comments. Please address those before merging.

File: app/src/main/kotlin/com/twain/interprep/presentation/ui/components/DropdownMenuInput.kt (hunk 5 above)
Comment (sarahhjchung) [reply app/src/main/kotlin/com/twain/interprep/presentation/ui/components/DropdownMenuInput.kt:19]: I'll make this change in a separate PR to unblock

**Pull request merged**